app/src/main/java/org/mozilla/fenix/components/TrackingProtectionPolicyFactory.kt

```diff
@@ -0,0 +1,89 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components
+
+import mozilla.components.concept.engine.EngineSession.TrackingProtectionPolicy
+import org.mozilla.fenix.utils.Settings
+
+/**
+ * Handles the logic behind creating new [TrackingProtectionPolicy]s.
+ */
+class TrackingProtectionPolicyFactory(private val settings: Settings) {
+
+    /**
+     * Constructs a [TrackingProtectionPolicy] based on current preferences.
+     *
+     * @param normalMode whether or not tracking protection should be enabled
+     * in normal browsing mode, defaults to the current preference value.
+     * @param privateMode whether or not tracking protection should be enabled
+     * in private browsing mode, default to the current preference value.
+     * @return the constructed tracking protection policy based on preferences.
+     */
+    @Suppress("ComplexMethod")
+    fun createTrackingProtectionPolicy(
+        normalMode: Boolean = settings.shouldUseTrackingProtection,
+        privateMode: Boolean = true
+    ): TrackingProtectionPolicy {
+        val trackingProtectionPolicy =
+            when {
+                settings.useStrictTrackingProtection -> TrackingProtectionPolicy.strict()
+                settings.useCustomTrackingProtection -> return TrackingProtectionPolicy.select(
```

Comment:
nit: Could we  extract the custom tracking protection policy into a new function